resolver-dns/src/main/java/io/netty/resolver/dns/DnsResolveContext.java

```diff
@@ -411,21 +444,21 @@ private boolean handleRedirect(
                 for (int i = 0; i < additionalCount; i++) {
                     final DnsRecord r = res.recordAt(DnsSection.ADDITIONAL, i);
 
-                    if (r.type() == DnsRecordType.A && !parent.supportsARecords() ||
+                    if (r.type() != question.type() ||
```

Comment:
dropping the redirect may have undesired results if other behavior in this class changes, or DNS server is not as expected so consider adding a comment here. IIUC we are assuming the following behavior:

- this DNS resolver will issue a query for each `A` and `AAAA` records if they are both supported.
- it is assumed the DNS server will only provide redirects of the same type as the query, and therefore it is safe to drop redirects where types don't match. In addition if the DNS server responds with mismatching (or maybe all types) types it is desirable to filter the redirects to avoid multiple queries for the same redirect.

Reply:
Actually, it was a bug. This change has to be reverted.